core/commons/che-core-commons-logback/src/main/java/org/eclipse/che/commons/logback/EnvironmentVariablesLogLevelPropagator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.commons.logback;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.LoggerContextListener;
+import ch.qos.logback.core.spi.ContextAwareBase;
+import ch.qos.logback.core.spi.LifeCycle;
+import java.util.Arrays;
+
+public class EnvironmentVariablesLogLevelPropagator extends ContextAwareBase
+    implements LoggerContextListener, LifeCycle {
+
+  private boolean isStarted;
+
+  private void setLoggerLevel(String loggerConfig) {
+    int i = loggerConfig.indexOf('=');
+    if (i < 0) {
+      return;
+    }
+    String loggerName = loggerConfig.substring(0, i);
+    String levelStr = loggerConfig.substring(i + 1);
+    if (loggerName == null) {
+      return;
+    }
+    if (levelStr == null) {
+      return;
+    }
+    loggerName = loggerName.trim();
+    levelStr = levelStr.trim();
+
+    addInfo("Trying to set level " + levelStr + " to logger " + loggerName);
+    LoggerContext lc = (LoggerContext) context;
+
+    Logger logger = lc.getLogger(loggerName);
+    if ("null".equalsIgnoreCase(levelStr)) {
+      logger.setLevel(null);
+    } else {
+      Level level = Level.toLevel(levelStr, null);
+      if (level != null) {
+        logger.setLevel(level);
+      }
+    }
+  }
+
+  @Override
+  public void start() {
+
+    String config = System.getenv("CHE_LOGGER_CONFIG");
+    if (config != null && !config.isEmpty()) {
+      Arrays.stream(config.split(",")).map(String::trim).forEach(this::setLoggerLevel);
```

Comment:
As far as I understand RuntimeException will be thrown in a case when `CHE_LOGGER_CONFIG` value has the wrong format. Am I right? If yes, maybe it would be better to catch it and log pretty message like '`CHE_LOGGER_CONFIG` env variable has the wrong format. It should be ...'

Reply:
No, AFAIK it would be just silently ignored. In fact, the same way if you put not existed classes or packages in logback.xml